### DISCUSSION TITLE
UI Issue: 'Add Training Data' Option Missing When No Default Training Data is Present

### DISCUSSION THREAD

**Discussion by anusreesoumya-nv** [vanna-ai/vanna#613] [Q&A]
The Vanna Flask app UI currently does not provide an option to add training data if the default training data is not present. Instead, it displays an error message: "No training data found, please add some training data first."

We would prefer to have an "Add Training Data" option available even when no default training data is present.

Is there a solution for this issue?

**Comment by zainhoda** [comment]
This behavior is intentional -- initial training on DDL/information schema should be done before launching the app in order to maximize database compatibility.

**Reply by anusreesoumya-nv** [reply]
@zainhoda ,While I understand the need for initial training on DDL/information schema before launching the app, I've encountered a situation where, after the default training data is deleted from the UI, there is no visible option to add training data again. This results in an error message: "No training data found, please add some training data first." 

Could we consider implementing an "Add Training Data" button or option in the UI for such cases? This would allow users to add new training data directly through the UI without requiring backend access or intervention, thereby improving the user experience.

**Comment by sinjup** [comment]
Hi, @zainhoda ,  I am also having a similar issue. After removing all the training data, I'm unable to add new training data because the UI doesn't provide an option for it. This creates a problem when all existing training data is removed, as it leaves no pathway to add new data. 
Could the UI be adjusted to always provide an option to add new training data, even when no existing data is present? Alternatively, could it prevent the deletion of the last available training data by displaying error messages in the UI?

**Reply by sqlquery** [reply]
Me also facing the same issue .......please provide an option for adding training data.
